build/Dockerfile

```diff
@@ -35,19 +35,22 @@ WORKDIR /build/windows_exporter/
 COPY windows_exporter/ .
 RUN GOOS=windows /build/promu/promu build -v
 
+# Build CNI plugins
+WORKDIR /build/containernetworking-plugins/
+COPY containernetworking-plugins/ .
+ENV CGO_ENABLED=0
+RUN ./build_windows.sh
+
 # Build Kubernetes node binaries
 WORKDIR /build/kubernetes/
 COPY kubernetes/ .
+RUN rm /build/kubernetes/.git
```

Comment:
This looks very odd. Yes, I understand you are trying to fix a cache miss but I think you are on the wrong track. You shouldn't have to do anything like this to fix caching.

Reply:
What i am doing here is, just copying the metadata needed for kubernetes repo. And not for other sub-modules. the `rm` is because, we cannot replace a file with directory with `COPY` instruction.